File: remove-comments/examples/javascript/calculator.js
/**
 * Calculator module for basic arithmetic operations
 * @author Developer
 * @version 1.0
 */

// Import required modules
const readline = require('readline');

/* 
   Multi-line comment explaining the purpose
   This calculator supports basic operations
*/
class Calculator {
    /**
     * Constructor for Calculator class
     */
    constructor() {
        this.history = []; // Store calculation history
    }

    // Single line comment before method
    add(a, b) {
        const result = a + b; // Inline comment
        /* Block comment in method */
        this.history.push(`${a} + ${b} = ${result}`);
        return result;
    }

    /*
     * Subtract two numbers
     * @param {number} a - First number
     * @param {number} b - Second number  
     * @returns {number} The difference
     */
    subtract(a, b) {
        const result = a - b;
        this.history.push(`${a} - ${b} = ${result}`);
        return result;
    }

    multiply(a, b) {
        // Calculate multiplication
        const result = a * b;
        return result; // Return the product
    }

    divide(a, b) {
        if (b === 0) {
            throw new Error("Division by zero"); // Error handling
        }
        /* Calculate division result */
        const result = a / b;
        return result;
    }

    // Method to display calculation history
    showHistory() {
        console.log("// Calculation History //");
        this.history.forEach(entry => {
            console.log(entry); // Print each entry
        });
    }
}

/* 
   Main function to demonstrate calculator usage
*/
function main() {
    const calc = new Calculator(); // Create calculator instance
    
    // Perform some calculations
    console.log(calc.add(5, 3)); // Should output 8
    console.log(calc.subtract(10, 4)); // Should output 6
    console.log(calc.multiply(7, 2)); // Should output 14
    
    /* Display results */
    calc.showHistory();
}

// Export the Calculator class
module.exports = Calculator;

// Run main function if this file is executed directly
if (require.main === module) {
    main(); // Execute main function
}

File: remove-comments/examples/go/calculator.go
//go:build ignore

package main

import (
	"fmt"
)

//go:generate

// Calculator represents a simple calculator
type Calculator struct {
	// result holds the current calculation result
	result float64
}

// NewCalculator creates a new calculator instance
func NewCalculator() *Calculator {
	return &Calculator{
		result: 0, // Initialize result to zero
	}
}

// Add performs addition operation
func (c *Calculator) Add(a, b float64) float64 {
	// Add two numbers and return the result
	c.result = a + b
	return c.result
}

// Subtract performs subtraction operation
func (c *Calculator) Subtract(a, b float64) float64 {
	c.result = a - b // Simple subtraction
	return c.result
}

/*
Multiply function performs multiplication
It takes two float64 parameters and returns their product
*/
func (c *Calculator) Multiply(a, b float64) float64 {
	c.result = a * b
	return c.result
}

/* Divide function performs division */
func (c *Calculator) Divide(a, b float64) float64 {
	if b == 0 {
		// Cannot divide by zero
		fmt.Println("Error: Division by zero")
		return 0
	}
	c.result = a / b /* Calculate division */
	return c.result
}

// GetResult returns the current result
func (c *Calculator) GetResult() float64 {
	return c.result // Return stored result
}

/*
Main function demonstrates calculator usage
This is a multi-line comment block
*/
func main() {
	calc := NewCalculator() // Create calculator instance

	// Perform some calculations
	fmt.Println("Addition:", calc.Add(10, 5))           // Should print 15
	fmt.Println("Subtraction:", calc.Subtract(10, 3))   // Should print 7
	fmt.Println("Multiplication:", calc.Multiply(4, 6)) // Should print 24
	fmt.Println("Division:", calc.Divide(20, 4))        // Should print 5

	/* Display final result */
	fmt.Printf("Final result: %.2f\n", calc.GetResult())
}
